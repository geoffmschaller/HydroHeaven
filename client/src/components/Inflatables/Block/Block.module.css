.block {
  padding: 0 5px;
}

/*# sourceMappingURL=Block.module.css.map */
