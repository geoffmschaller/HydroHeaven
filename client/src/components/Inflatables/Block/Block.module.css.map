{"version":3,"sourceRoot":"","sources":["Block.module.sass"],"names":[],"mappings":"AACA;EACC","file":"Block.module.css"}