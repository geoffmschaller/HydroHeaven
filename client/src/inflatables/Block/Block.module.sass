@import "src/static/sass/config"

.block
	padding: 0 5px