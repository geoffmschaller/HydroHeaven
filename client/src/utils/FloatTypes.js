export const FLOAT_LEFT = 'left';
export const FLOAT_RIGHT = 'right';