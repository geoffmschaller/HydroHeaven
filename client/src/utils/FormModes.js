export const FORM_READY = 'form_ready';
export const FORM_SUBMITTING = 'form_submitting';
export const FORM_SUCCESS = "form_sucess";