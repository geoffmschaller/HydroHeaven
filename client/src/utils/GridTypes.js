export const RANDOM_GRID = 'random_grid';
export const ALL_GRID = 'all_grid';