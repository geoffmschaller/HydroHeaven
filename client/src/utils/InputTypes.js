export const LINE_INPUT = 'line_input';
export const PASSWORD_INPUT = 'password_input';
export const MULTI_LINE_INPUT = 'multi_line_input';