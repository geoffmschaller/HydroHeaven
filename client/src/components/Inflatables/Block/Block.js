import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styles from './Block.module.css';
import { FLOAT_LEFT, FLOAT_RIGHT } from '../../../utils/FloatTypes';

class Block extends Component {
	constructor(props) {
		super(props);
		this.state = {
			windowWidth: 0,
		};

		this.calculateWidth = () => {
			this.setState({ windowWidth: window.innerWidth });
		};
	}

	componentDidMount() {
		this.calculateWidth();
		window.addEventListener('resize', this.calculateWidth);
	}

	componentWillUnmount() {
		window.removeEventListener('resize', this.calculateWidth);
	}

	render() {
		const width = this.state.windowWidth >= 400 ? `${this.props.width}%` : '100%';
		return (
			<div className={styles.block} style={{ float: this.props.float, width }}>{this.props.children}</div>
		);
	}
}

Block.propTypes = {
	float: PropTypes.oneOf([FLOAT_LEFT, FLOAT_RIGHT]).isRequired,
	width: PropTypes.number.isRequired,
};

export default Block;
