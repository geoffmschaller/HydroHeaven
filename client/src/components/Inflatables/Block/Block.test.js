import React from 'react';
import Block from './Block';
import {FLOAT_LEFT} from "../../../utils/FloatTypes";

it("Block Component Test", () => {

	const wrapper = shallow(
		<Block float={FLOAT_LEFT} width={30}>
			<p>I AM INSIDE OF A BLOCK</p>
		</Block>
	);
	expect(wrapper).toMatchSnapshot();

});