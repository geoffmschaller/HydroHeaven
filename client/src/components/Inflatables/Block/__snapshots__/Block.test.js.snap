// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Block Component Test 1`] = `
<div
  className="block"
  style={
    Object {
      "float": "left",
      "width": "30%",
    }
  }
>
  <p>
    I AM INSIDE OF A BLOCK
  </p>
</div>
`;
